test/modules/deitz/test_nested_modules2.chpl

```diff
@@ -3,9 +3,10 @@ module OuterModule {
     var x: int = 2;
   }
 
-  use M1;
+  public use M1;
```

Comment:
I don't think this should have to be 'public use'. If it doesn't work if it is not, let's add a future / issue about it.

Reply:
That doesn't seem right to me:  `use` is `private` / non-transitive by default, so this one permits OuterModule to refer to M1.x, but not for things that `use OuterModule` to do so.  I don't think we've discussed making inner modules any different w.r.t. these rules as other modules.

If M2 had a `use OuterModule.M1;` or `use super.M1;` then it wouldn't need to be public.

Reply:
>  I don't think we've discussed making inner modules any different w.r.t. these rules as other modules

Right, I think we should discuss that.

Reply:
OK, I think you'd better write up the issue then.

Reply:
Hmm, weird...  It seems we're inconsistent depending on whether the symbol in question is a function or a variable (or a mix).  I'm filing some futures in my PR about this.

Reply:
See https://github.com/chapel-lang/chapel/pull/15290/commits/e691c2c1b60209a737e0c4a00713133d792d4bf5

Reply:
Ohhhh... The reason we're inconsistent about functions and variables is that this PR's changes only affect scopeResolve.cpp and not function visibility.  That requires a symmetric effort.  Looking into it now, but won't be merging today.